### PR TITLE
feat: Allow overriding response MSH content #103

### PR DESCRIPTION
**Pull request by jtarvainen**
## Issue

https://github.com/Bugs5382/node-hl7-server/issues/103

## Search terms

MSH, override

## Questioner

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

Other things:

- [x] Write a unit test or update unit tests that cover your change in the code?
- [x] Set the PR to merge into the develop branch?
- [x] Clear documentation per the guidelines in the [README.md](../../README.md) as we are support medical applications?

### CODE REVIEW

**Comment by jtarvainen** [comment]
I replaced the existing `overrideMSH` parameter (boolean), which was only used to set the MSH.9.3 value, with a comprehensive `mshOverrides` parameter (object) that allows overriding any MSH field value. This is a breaking change, but to me it makes sense to handle all possible MSH overrides with a single parameter.

So, to me, this change would bump the version to 3.0.0, but I didn't update the version before discussing it here.

**Comment by Bugs5382** [comment]
@jtarvainen Hello! It's my normal Saturday to code and checking this out!

File: src/server/modules/sendResponse.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import {
   HL7_2_5_1, HL7_2_6, HL7_2_7, HL7_2_7_1, HL7_2_8
 } from 'node-hl7-client/hl7'
 import { PROTOCOL_MLLP_FOOTER, PROTOCOL_MLLP_HEADER } from '../../utils/constants.js'
+import type { ListenerOptions } from '../../utils/normalize'
```
Comment (Bugs5382) [inline src/server/modules/sendResponse.ts:14]: @jtarvainen  just needs to change to .js, but I will do that in the backend during merge

**Pull request merged**